File: temas/anotaciones/Java/BankAccount.java
import java.util.*;
import java.io.*;
import java.time.*;

public final class BankAccount implements Comparable<BankAccount> {
  private final String id;
  @LocalDateNow
  private LocalDate creationDate;
  @CompareByDate
  private Comparator comparator;

  public BankAccount(String number) {
    this.id = number;
    comparator = new BankAccountComparatorById();
  }

  public LocalDate getCreationDate() {
    return creationDate;
  }

  public void setCreationDate(LocalDate date) {
    this.creationDate = date;
  }

  public String getId() {
    return id;
  }

  @CompareById
  public void setComparator(Comparator cmp) {
    comparator = cmp;
  }

  @Override
  public int compareTo(BankAccount other) {
    if (this == other)
      return 0;
    assert this.equals(other) : "compareTo inconsistent with equals.";
    return comparator.compare(this, other);
  }

  @Override
  public boolean equals(Object other) {
    if (this == other)
      return true;
    if (!(other instanceof BankAccount))
      return false;
    BankAccount that = (BankAccount) other;
    return this.id.equals(that.getId());
  }

  @Override
  public String toString() {
    return id.toString();
  }
}

File: temas/anotaciones/ruby/BankAccount.ts

// BankAccount.ts

import { ComparableDecorator } from './Comparable';

@ComparableDecorator
export class BankAccount {
    private readonly id: string;
    private creationDate: Date;

    constructor(id: string) {
        this.id = id;
    }

    getCreationDate(): Date {
        return this.creationDate;
    }

    setCreationDate(creationDate: Date): void {
        this.creationDate = creationDate;
    }

    getId(): string {
        return this.id;
    }
}